### PR TITLE
MINIFICPP-2429 Remove onTriggerSharedPtr/onScheduleSharedPtr

### PR DESCRIPTION
**Pull request by martinzink**
MINIFICPP-2430 Remove CollectorInitiatedSubscription

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/lua/LuaProcessSession.cpp [hunk 1]
```diff
@@ -43,78 +39,58 @@ std::shared_ptr<LuaScriptFlowFile> LuaProcessSession::get() {
...
   });
 }
 
 std::shared_ptr<LuaScriptFlowFile> LuaProcessSession::create() {
-  if (!session_) {
-    throw std::runtime_error("Access of ProcessSession after it has been released");
-  }
-
-  auto result = std::make_shared<LuaScriptFlowFile>(session_->create());
+auto result = std::make_shared<LuaScriptFlowFile>(session_.create());
```
Comment (szaszm) [inline extensions/lua/LuaProcessSession.cpp:82]: indentation problem

Comment (martinzink) [reply extensions/lua/LuaProcessSession.cpp]: good catch https://github.com/apache/nifi-minifi-cpp/pull/1843/commits/434975b6341f9ba04e35e07d2b0559ca7b87846f#diff-f980ccede23f6ad9af2a715b01315fabe0b5dcb06d2f75b948123453e4fe6634R82

File: extensions/python/types/PyProcessSession.h [hunk 2]
```diff
@@ -27,7 +27,7 @@ namespace org::apache::nifi::minifi::extensions::python {
 
 class PyProcessSession {
  public:
-  explicit PyProcessSession(std::shared_ptr<core::ProcessSession> session);
+  explicit PyProcessSession(gsl::not_null<core::ProcessSession*> session);
```
[the remark this answers is not in the document]
Comment (martinzink) [reply extensions/python/types/PyProcessSession.h]: good idea, https://github.com/apache/nifi-minifi-cpp/pull/1843/commits/434975b6341f9ba04e35e07d2b0559ca7b87846f#diff-c6e6761718cb79a8c101319d280e709c261cf4231184df8fe7c88e68fc432e6dR32

**Review by lordgamez — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Comment by szaszm** [comment]
rebased to main

**Pull request closed without merging**